### ISSUE TITLE
MongoDbFixture fails tests with NullReferenceException

### ISSUE THREAD

**Issue by Xeevis** [abpframework/abp#4153]
I've encountered a curious problem in my project that I'm sadly unable to reproduce with fresh templates, but someone here might have a clue, as I don't want to spend whole day researching this. (Also it might help in case I'm not alone)

Problem lies in MongoDB template test I've implemented without any changes into my existing solution that is already highly modified former template. This line will throw NullReferenceException.
https://github.com/abpframework/abp/blob/bfce4c2752d98cc98e693e726ceb67b287140412/templates/app/aspnet-core/test/MyCompanyName.MyProjectName.MongoDB.Tests/MongoDb/MongoDbFixture.cs#L9

![image](https://user-images.githubusercontent.com/5835044/83337802-8e1d6b00-a2be-11ea-8713-222b00c35ba8.png)

It's curious because if I change that public field into property.
```csharp
//public static readonly string ConnectionString = MongoDbRunner.ConnectionString;
public static string ConnectionString => MongoDbRunner.ConnectionString;
```
It suddenly works fine ...

![image](https://user-images.githubusercontent.com/5835044/83337812-a2616800-a2be-11ea-824d-1270f92187c9.png)

With fresh template, this change isn't necessary, it just works with field giving me proper headache 😕. Wouldn't property make more sense there though?

ABP 2.8.0
.NET Core SDK 3.1.300

```
Message: 
    Volo.Abp.AbpInitializationException : An error occurred during ConfigureServices phase of the module MyProject.MongoDB.MyProjectMongoDbTestModule, MyProject.MongoDB.Tests, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null. See the inner exception for details.
    ---- System.TypeInitializationException : The type initializer for 'MyProject.MongoDB.MongoDbFixture' threw an exception.
    -------- System.NullReferenceException : Object reference not set to an instance of an object.
  Stack Trace: 
    ModuleLoader.ConfigureServices(List`1 modules, IServiceCollection services)
    ModuleLoader.LoadModules(IServiceCollection services, Type startupModuleType, PlugInSourceList plugInSources)
    AbpApplicationBase.LoadModules(IServiceCollection services, AbpApplicationCreationOptions options)
    AbpApplicationBase.ctor(Type startupModuleType, IServiceCollection services, Action`1 optionsAction)
    AbpApplicationWithExternalServiceProvider.ctor(Type startupModuleType, IServiceCollection services, Action`1 optionsAction)
    AbpApplicationFactory.Create(Type startupModuleType, IServiceCollection services, Action`1 optionsAction)
    AbpApplicationFactory.Create[TStartupModule](IServiceCollection services, Action`1 optionsAction)
    ServiceCollectionApplicationExtensions.AddApplication[TStartupModule](IServiceCollection services, Action`1 optionsAction)
    AbpIntegratedTest`1.ctor()
    MyProjectTestBase`1.ctor()
    MyProjectApplicationTestBase.ctor()
    SampleAppServiceTests.ctor() line 18
    ----- Inner Stack Trace -----
    MyProjectMongoDbTestModule.ConfigureServices(ServiceConfigurationContext context) line 16
    ModuleLoader.ConfigureServices(List`1 modules, IServiceCollection services)
    ----- Inner Stack Trace -----
    MongoDbFixture.cctor() line 8
```

**Comment by maliming** [comment]
It's strange, can you share a project that reproduces the problem?

**Comment by Xeevis** [comment]
Bingo! It indeed is caused by the order of fields, why didn't I think of that?! The reason why I encountered this is quite simple too, I'm using the popular [CodeMaid VS extension](https://marketplace.visualstudio.com/items?itemName=SteveCadwallader.CodeMaid) that by default sorts member fields alphabetically and never have I encountered this issue before.

This leads me to believe this is highly unusual pattern for class member fields we have here. In my opinion this shouldn't even make it through the compiler. While member field declarations are order agnostic and therefore compiler won't complain if you reference one from another in any order, it's unintuitive code since initialization is still ordered sequence.

Like in this example, one would totally expect either compiler to throw, and if not, then some output from the `FooBar`. And yet neither will happen ...
```csharp
private static readonly string FooBar = Foo + Bar;
private static readonly string Foo = "Foo";
private static readonly string Bar = "Bar";
static void Main()
{
    Console.WriteLine("Hello " + FooBar); // Outputs "Hello "
}
```

It would probably be better to go with properties to reduce surface area for such bugs ...
